jni/src/test/java/com/horizen/commitmenttreenative/ScCommitmentCertPathTest.java

```diff
@@ -156,4 +160,90 @@ public void completeTest() {
             }
         }
     }
+
+    @Test
+    public void updatePath() {
+        try (
+                CommitmentTree commTreeReal = CommitmentTree.init();
+                FieldElement scId1 = FieldElement.createFromLong(10);
+                FieldElement scId2 = FieldElement.createFromLong(20);
+                FieldElement certLeaf11 = FieldElement.createRandom();
+                FieldElement certLeaf12 = FieldElement.createRandom();
+                FieldElement certLeaf21 = FieldElement.createRandom();) {
+
+            byte[] scId1Bytes = scId1.serializeFieldElement();
+            byte[] scId2Bytes = scId2.serializeFieldElement();
+            FieldElement[] certs1 = { certLeaf11, certLeaf12 };
+            FieldElement[] certs2 = { certLeaf21 };
+            ArrayList<byte[]> certs1HashBytes = Arrays.stream(certs1)
+                    .map(FieldElement::serializeFieldElement)
+                    .collect(Collectors.toCollection(ArrayList::new));
+
+            ArrayList<byte[]> certs2HashBytes = Arrays.stream(certs2)
+                    .map(FieldElement::serializeFieldElement)
+                    .collect(Collectors.toCollection(ArrayList::new));
+
+            for (byte[] h : certs1HashBytes) {
+                commTreeReal.addCertLeaf(scId1Bytes, h);
+            }
+            for (byte[] h : certs2HashBytes) {
+                commTreeReal.addCertLeaf(scId2Bytes, h);
+            }
+
+            try (
+                    FieldElement expectedScTxCommitmentRoot = commTreeReal.getCommitment().get();
+                    MerklePath correctScPath = commTreeReal.getScCommitmentMerklePath(scId1Bytes).get();
+                    CommitmentTree commTreeFake = CommitmentTree.init();) {
```

Comment:
can we rename commTreeFake in commTreePartial ? since it will the way it will work on sdk implementation for real :-)

Reply:
Ok... I'll do it